### PR TITLE
Task-15: Refactor to typescript

### PR DESCRIPTION
**Pull request by svetamint**
```
Acceptance criteria: https://godelonline.sharepoint.com/Functions/LearningAndDevelopment/SitePages/Node---Proxy-Server.aspx#task-15-refactor-to-typescript

Type script is added and configured
Implement build and run with TS support
All files are refactored to TS
Types and models are covered with types
```

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const app: Express = express()
+
+app.use(express.json())
+app.use(express.urlencoded({ extended: true }))
+
+app.use(meteorRouter)
+
+app.use(errorHandler)
+
```
Comment (Mikhail1992) [inline index.ts:15]: we don't have this task but let's add pageNotFound

File: task9/config/config.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  port: number
+  startDate: string
+  endDate: string
+  nasaMeteorUrl: string
+  nasaRoverUrl: string
+  sol: number
+}
+
+const config: Config = {
+  port: parseInt(process.env.APP_PORT || '4000'),
```
Comment (Mikhail1992) [inline task9/config/config.ts:15]: `parseInt(process.env.APP_PORT) || 4000` looks better

File: task9/config/config.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  startDate: process.env.START_DATE || '2024-10-10',
+  endDate: process.env.END_DATE || '2024-10-11',
+  nasaMeteorUrl: `${process.env.NASA_API_URL}?api_key=${process.env.NASA_API_KEY}`,
+  nasaRoverUrl: process.env.NASA_ROVER_API_URL || '',
+  sol: parseInt(process.env.SOL || '1000')
```
Comment (Mikhail1992) [inline task9/config/config.ts:20]: same one

File: task9/delivery/meteorController.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          )
+        )
+      } else {
+        next(
+          new Exception(
+            StatusCodes.INTERNAL_SERVER_ERROR,
+            'An unknown error occurred'
+          )
+        )
+      }
```
Comment (Mikhail1992) [inline task9/delivery/meteorController.ts:61]: maybe better to leave only `next(error)` and then move this logic into errorHandler

File: task9/delivery/meteorController.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
```
Comment (Mikhail1992) [inline task9/delivery/meteorController.ts:97]: same one

**Review by mdziarkach — COMMENTED** [review]
If you start redoing it for classes, then you need to do it right
I shared with you a link to a finished implementation of another project

File: src/app.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+  useRoutes(): void {
+    this.app.use(this.meteorController.router)
+  }
+
+  public async init(): Promise<void> {
+    this.useMiddleware()
+    this.useRoutes()
+    this.useExceptionFilters()
+    this.configureNunjucks()
```
Comment (Mikhail1992) [inline src/app.ts:57]: this.useMiddleware()
this.configureNunjucks()
this.useRoutes()
this.useExceptionFilters()

File: index.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+nunjucks.configure('views', {
+  autoescape: true,
+  express: app
+})
+
+app.listen(config.port, () => {
+  console.log(`Server is running on port ${config.port}`)
+})
```
Comment (Mikhail1992) [inline index.ts:23]: looks like you can remove this file

File: task9/config/config.ts (hunk 2 above)
Comment (Mikhail1992) [reply task9/config/config.ts:15]: do you need this file?

**Review by mdziarkach — COMMENTED** [review]
Judging by the structure, you have 2 solutions at once. Maybe you can delete the old approach and leave only the new one?

File: task9/config/config.ts (hunk 2 above)
Comment (svetamint) [reply task9/config/config.ts:15]: no, we don't and I'm not using it, that's why I asked you check specific commit )

File: src/app.ts (hunk 6 above)
Comment (svetamint) [reply src/app.ts]: fixed

**Review by mdziarkach — APPROVED** [review]

**Pull request merged**